src/AutoRest.CSharp/Common/Output/Models/Types/SchemaObjectType.cs

```diff
@@ -32,65 +31,51 @@ namespace AutoRest.CSharp.Output.Models.Types
     internal abstract class SchemaObjectType : SerializableObjectType
     {
         private readonly SerializationBuilder _serializationBuilder;
+        private readonly InputModelTypeUsage _usage;
         private readonly TypeFactory _typeFactory;
-        private readonly OutputLibrary? _library;
-        private readonly SchemaTypeUsage _usage;
-
-        private readonly IReadOnlyList<KnownMediaType> _supportedSerializationFormats;
 
         private ObjectTypeProperty? _additionalPropertiesProperty;
         private CSharpType? _implementsDictionaryType;
 
-        protected SchemaObjectType(ObjectSchema objectSchema, string defaultNamespace, TypeFactory typeFactory, SchemaUsageProvider schemaUsageProvider, OutputLibrary? library, SourceInputModel? sourceInputModel)
+        protected SchemaObjectType(InputModelType inputModel, string defaultNamespace, TypeFactory typeFactory, SourceInputModel? sourceInputModel, SerializableObjectType? defaultDerivedType = null)
```

Comment:
I think this is our first step of consolidation? By only modifying `SchemaObjectType` to consume input types instead of schemas.
I assume we should have a next step that we unify these two classes of `ModelTypeProvider` and `SchemaObjectType`, is this true?

Reply:
That is true